src/Ledger/Transaction.lagda

```diff
@@ -35,11 +35,15 @@ the transaction body are:
   \item The size and the hash of the serialized form of the transaction that was included in the block.
 \end{itemize}
 
+Finally, $\fun{txid}$ computes the transaction id of a given transaction.
+This function must produce a unique id for each unique transaction body.
+\textbf{We assume that} $\fun{txid}$ \textbf{is injective.}
+
 \begin{figure*}[h]
 \emph{Abstract types}
-\AgdaTarget{Ix, TxId, Epoch, AuxiliaryData}
+\AgdaTarget{Ix, TxId, Epoch, AuxiliaryData, PolicyId}
 \begin{code}
-        Ix TxId AuxiliaryData : Set
+        Ix TxId AuxiliaryData PolicyId AssetName : Set
```

Comment:
These two types are unnecessary

Reply:
Just to confirm, I'll remove the `PolicyId` parameter from `TransactionStructure` and, in 

```agda
module Ledger.Utxo.Properties (txs : TransactionStructure) where
```

I'll import `TokenAlgebra` as follows:

```agda
open import Ledger.TokenAlgebra ScriptHash
```

instead of, say,

```agda
open import Ledger.TokenAlgebra (PolicyId txs)
```

Is that right?